tests/light/functional_tests/destination_drivers/clickhouse_destination/test_clickhouse_complex_types.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python
+#############################################################################
+# Copyright (c) 2025 Axoflow
+# Copyright (c) 2025 Andras Mitzki <andras.mitzki@axoflow.com>
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+import uuid
+
+from axosyslog_light.common.file import copy_shared_file
+
+
+custom_input_msg = f"test message {str(uuid.uuid4())}"
+clickhouse_options = {
+    "database": "default",
+    "table": "test_table",
+    "user": "default",
+    "password": "'password'",
+}
+
+
+def configure_syslog_ng_with_clickhouse_dest(config):
+    generator_source = config.create_example_msg_generator_source(num=1, template=f'"{custom_input_msg}"')
+    clickhouse_options_copy = clickhouse_options.copy()
+    filterx = config.create_filterx('''
+    message = {
```

Comment:
Testing more complex types, like, datetime, datetime64, json, would be really useful.